Controller/Webhook/Update.php

```diff
@@ -0,0 +1,197 @@
+    protected $logitrail;
+
+    /** @var \Magento\Framework\Logger\Monolog  */
+    protected $logger;
+
+    /** @var \Magento\Framework\App\Config\ScopeConfigInterface  */
+    protected $scopeConfig;
+
+    /** @var \Magento\CatalogInventory\Api\StockRegistryInterface  */
+    protected $stockRegistry;
+
+    /** @var \Magento\Sales\Model\OrderRepository  */
+    protected $orderRepository;
+
+    /** @var \Magento\Sales\Model\Convert\OrderFactory  */
+    protected $convertOrderFactory;
+
+    /** @var \Magento\Sales\Model\Order\Shipment\TrackFactory  */
+    protected $trackFactory;
+
+    /** @var \Magento\Shipping\Model\ShipmentNotifier  */
+    protected $shipmentNotifier;
+
+    /** @var \Magento\Framework\DB\TransactionFactory  */
+    protected $transactionFactory;
+
+    public function __construct(
+        \Codaone\LogitrailModule\Model\Logitrail $logitrail,
+        \Magento\Backend\App\Action\Context $context,
+        \Magento\Framework\Logger\Monolog $logger,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\CatalogInventory\Api\StockRegistryInterface $stockRegistry,
+        \Magento\Sales\Model\OrderRepository $orderRepository,
+        \Magento\Sales\Model\Convert\OrderFactory $convertOrderFactory,
+        \Magento\Sales\Model\Order\Shipment\TrackFactory $trackFactory,
+        \Magento\Shipping\Model\ShipmentNotifier $shipmentNotifier,
+        \Magento\Framework\DB\TransactionFactory $transactionFactory
+
+    ) {
+        parent::__construct($context);
+        $this->logitrail = $logitrail;
+        $this->logger = $logger;
+        $this->scopeConfig = $scopeConfig;
+        $this->stockRegistry = $stockRegistry;
+        $this->orderRepository = $orderRepository;
+        $this->convertOrderFactory = $convertOrderFactory;
+        $this->trackFactory = $trackFactory;
+        $this->shipmentNotifier = $shipmentNotifier;
+        $this->transactionFactory = $transactionFactory;
+    }
+
+    /*
+     *  Handle webservice request
+     */
+    public function execute()
+    {
+        /** @var \Magento\Framework\App\Request\Http $request */
+        $request = $this->getRequest();
+
+        /** @var \Zend\Http\Headers $header */
+        $header = $request->getHeaders();
+
+        /** @var bool $result */
+        $result = false;
+
+        if ($this->authenticate($header)) {
+            $content = $request->getContent();
+
+            $data = $this->logitrail->getApi()->processWebhookData($content);
+
+            $this->logger->debug(json_encode($data, JSON_UNESCAPED_UNICODE));
+
+            switch ($data["event_type"]) {
+                case "product.inventory.change":
+                    $result = $this->handleInventoryChange($data);
+                    break;
+                case "order.shipped":
+                    $result = $this->handleOrderShipped($data);
+                    break;
+                default:
+                    $result = true;
```

Comment:
The response shouldn't be "200 OK" when we don't handle the webhook at all. Now the sending end thinks everything went ok and the Magento 2 installation using this plugin has the info that was part of the webhook request.